File: BPG_CMS_BE/src/BPG.Domain/Entities/Supplier.cs
namespace BPG.Domain.Entities;

public class Supplier : BaseEntity
{
    public long SupplierId { get; set; }
    public string SupplierName { get; set; } = string.Empty;
    public string? ContactInfo { get; set; }
    public string? Address { get; set; }
    public string? ServiceArea { get; set; }
    public decimal? Rating { get; set; }
    public string? EvaluationNote { get; set; }
    public string CollaborationStatus { get; set; } = BPG.Domain.Constants.CollaborationStatus.Regular;

    public ICollection<PurchaseOrder> PurchaseOrders { get; set; } = new List<PurchaseOrder>();
}

File: BPG_CMS_FE/src/types/supplier.ts
export interface Supplier {
  supplierId: number;
  supplierName: string;
  contactInfo?: string;
  address?: string;
  serviceArea?: string;
  rating?: number;
  evaluationNote?: string;
  collaborationStatus: 'Strategic' | 'Regular' | 'Restricted' | 'Blacklisted';
}

export interface GetSuppliersQuery {
  pageNumber?: number;
  pageSize?: number;
  search?: string;
  sortBy?: string;
  sortDescending?: boolean;
  collaborationStatus?: string;
}

export interface ImportSuppliersResult {
  successCount: number;
  skippedCount: number;
  errors: string[];
  importedSuppliers: Supplier[];
}
